### PR TITLE
Extract Terraform outputs to output-var with shell flavors

### PR DESCRIPTION
**Pull request by gaelL**
Refactoring of https://github.com/cycloidio/docker-cycloid-toolkit/pull/39

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -219,6 +219,14 @@ shared:
 ## extract-terraform-outputs
 
 This script is mostly expected to be used by the `merge-stack-and-config` script.
+It will creates a YAML and a Shell files from terraform outputs. In addition it will also export all Terraform outputs as environment variables.
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
Its purpose is to export all terraform outputs as both a YAML and shell script files in addition to loading them as environment variables in the current shell execution scope.
```

File: README.md [hunk 2]
```diff
@@ -219,6 +219,14 @@ shared:
...
+
+./scripts/extract-terraform-outputs
+  * `(TERRAFORM_METADATA_FILE)` Default `terraform/metadata`.'
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * `(TERRAFORM_METADATA_FILE)` Defaults to `terraform/metadata`.'
```

File: README.md [hunk 3]
```diff
@@ -219,6 +219,14 @@ shared:
...
+  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Default `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Defaults to `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
```

I don't understand the last part.

File: README.md [hunk 4]
```diff
@@ -219,6 +219,14 @@ shared:
...
+  * `(OUTPUT_ENV_VAR_FILE)` Shell environment variables file. Default `output-var/env`.'
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * `(OUTPUT_ENV_VAR_FILE)` Shell environment variables file. Defaults to `output-var/env`.'
```

File: README.md [hunk 5]
```diff
@@ -219,6 +219,14 @@ shared:
...
+  * `(OUTPUT_VAR_PATH)` base path used for all *_VAR_FILE. Default `output-var`.'
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * `(OUTPUT_VAR_PATH)` base path used for all *_VAR_FILE. Defaults to `output-var`.'
```

File: scripts/extract-terraform-outputs [hunk 6]
```diff
@@ -6,7 +6,9 @@ usage()
     echo ''
     echo "$0"
     echo '  * `(TERRAFORM_METADATA_FILE)` Default `terraform/metadata`.'
-    echo '  * `(OUTPUT_ANSIBLE_VAR_FILE)` Default `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
+    echo '  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Default `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
```
Comment (sdurrheimer) [inline scripts/extract-terraform-outputs]:
```suggestion
    echo '  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Defaults to `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
```

same what does the last part mean?

File: scripts/extract-terraform-outputs [hunk 7]
```diff
@@ -6,7 +6,9 @@ usage()
     echo ''
     echo "$0"
     echo '  * `(TERRAFORM_METADATA_FILE)` Default `terraform/metadata`.'
```
Comment (sdurrheimer) [inline scripts/extract-terraform-outputs]:
```suggestion
    echo '  * `(TERRAFORM_METADATA_FILE)` Defaults to `terraform/metadata`.'
```

File: scripts/extract-terraform-outputs [hunk 8]
```diff
@@ -6,7 +6,9 @@ usage()
...
-    echo '  * `(OUTPUT_ANSIBLE_VAR_FILE)` Default `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
+    echo '  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Default `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
+    echo '  * `(OUTPUT_ENV_VAR_FILE)` Shell environment variables file. Default `output-var/env`.'
```
Comment (sdurrheimer) [inline scripts/extract-terraform-outputs]:
```suggestion
    echo '  * `(OUTPUT_ENV_VAR_FILE)` Shell environment variables file. Defaults to `output-var/env`.'
```

File: scripts/extract-terraform-outputs [hunk 9]
```diff
@@ -6,7 +6,9 @@ usage()
...
+    echo '  * `(OUTPUT_VAR_PATH)` base path used for all *_VAR_FILE. Default `output-var`.'
```
Comment (sdurrheimer) [inline scripts/extract-terraform-outputs]:
```suggestion
    echo '  * `(OUTPUT_VAR_PATH)` base path used for all *_VAR_FILE. Defaults to `output-var`.'
```

File: scripts/extract-terraform-outputs [hunk 10]
```diff
@@ -15,11 +17,27 @@ if [ -n "$1" ]; then usage; fi
...
 
 set -e
 
+if ! [ -f "$TERRAFORM_METADATA_FILE" ]; then
+    echo "$0 - ERROR: the Terraform output file '${TERRAFORM_METADATA_FILE}' does not exist."
+    exit 1
+fi
+
 mkdir -p $(dirname $OUTPUT_ANSIBLE_VAR_FILE)
+echo "############ Extracting Terraform output YAML -> $OUTPUT_ANSIBLE_VAR_FILE"
```
Comment (sdurrheimer) [inline scripts/extract-terraform-outputs]:
```suggestion
echo "############ Extracting Terraform outputs as YAML -> $OUTPUT_ANSIBLE_VAR_FILE"
```

File: scripts/extract-terraform-outputs [hunk 11]
```diff
@@ -15,11 +17,27 @@ if [ -n "$1" ]; then usage; fi
...
 python -c 'import sys, yaml, json; yaml.safe_dump(json.load(sys.stdin), sys.stdout, default_flow_style=False)' < $TERRAFORM_METADATA_FILE | tee -a $OUTPUT_ANSIBLE_VAR_FILE
 
-exit $?
+mkdir -p $(dirname $OUTPUT_ENV_VAR_FILE)
+echo "############ Extracting Terraform output SHELL -> $OUTPUT_ENV_VAR_FILE"
```
Comment (sdurrheimer) [inline scripts/extract-terraform-outputs]:
```suggestion
echo "############ Extracting Terraform outputs as a shell script -> $OUTPUT_ENV_VAR_FILE"
```

**Review by sdurrheimer — CHANGES_REQUESTED** [review]

**Comment by gaelL** [comment]
Changed applied, tests fixed and looks ok in the CI

File: scripts/extract-terraform-outputs [hunk 12]
```diff
@@ -5,8 +5,10 @@ usage()
     echo 'You should provide the following env vars :'
     echo ''
     echo "$0"
-    echo '  * `(TERRAFORM_METADATA_FILE)` Default `terraform/metadata`.'
-    echo '  * `(OUTPUT_ANSIBLE_VAR_FILE)` Default `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
+    echo '  * `(TERRAFORM_METADATA_FILE)` Defaults to `terraform/metadata`.'
+    echo '  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Defaults to `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
```
Comment (sdurrheimer) [inline scripts/extract-terraform-outputs]:
```suggestion
    echo '  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Defaults to `output-var/all`. You might want to use `ansible-playbook/group_vars/all`.'
```

File: README.md [hunk 13]
```diff
@@ -219,6 +219,14 @@ shared:
 ## extract-terraform-outputs
 
 This script is mostly expected to be used by the `merge-stack-and-config` script.
+Its purpose is to export all terraform outputs as both a YAML and shell script files in addition to loading them as environment variables in the current shell execution scope.
+
+./scripts/extract-terraform-outputs
+  * `(TERRAFORM_METADATA_FILE)` Defaults to `terraform/metadata`.'
+  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Defaults to `output-var/all`. You might expect provide `ansible-playbook/group_vars/all`.'
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * `(OUTPUT_ANSIBLE_VAR_FILE)` Ansible variables file. Defaults to `output-var/all`. You might want to use `ansible-playbook/group_vars/all`.'
```

**Review by sdurrheimer — CHANGES_REQUESTED** [review]

**Review by sdurrheimer — APPROVED** [review]

**Pull request merged**